### PR TITLE
Rename ProviderService to GitHubProviderService

### PR DESCRIPTION
**Pull request by dmjb**
Since this logic is Github-specific, name it accordingly, and move it under the Github sub package.

# Summary

***Provide a brief overview of the changes and the issue being addressed. 
Explain the rationale and any background necessary for understanding the changes. 
List dependencies required by this change, if any.***

Fixes #(related issue)

## Change Type

***Mark the type of change your PR introduces:***

- [ ] Bug fix (resolves an issue without affecting existing features)
- [ ] Feature (adds new functionality without breaking changes)
- [ ] Breaking change (may impact existing functionalities or require documentation updates)
- [ ] Documentation (updates or additions to documentation)
- [x] Refactoring or test improvements (no bug fixes or new functionality)

# Testing

***Outline how the changes were tested, including steps to reproduce and any relevant configurations. 
Attach screenshots if helpful.***

# Review Checklist:

- [x] Reviewed my own code for quality and clarity.
- [x] Added comments to complex or tricky code sections.
- [ ] Updated any affected documentation.
- [ ] Included tests that validate the fix or feature.
- [ ] Checked that related changes are merged.

### CODE REVIEW

File: internal/controlplane/handlers_oauth.go [hunk 1]
```diff
@@ -613,7 +614,7 @@ func (s *Server) VerifyProviderCredential(ctx context.Context,
 // cannot be tied back to a specific project, so we create a new project for the provider.
 //
 // This is a callback because we want to encapsulate components like s.cfg.Identity,
-// s.marketplace, s.authzClient and the like from the the providers implementation.
```
Comment (eleftherias) [inline internal/controlplane/handlers_oauth.go:616]: nit: GoLand is bad at renaming

File: internal/controlplane/handlers_providers.go [hunk 2]
```diff
@@ -71,7 +71,7 @@ func (s *Server) GetProvider(ctx context.Context, req *minderv1.GetProviderReque
 	}, nil
 }
 
-// ListProviders lists the providers available in a specific project.
```
Comment (eleftherias) [inline internal/controlplane/handlers_providers.go:74]: nit: More renaming in comments

File: internal/controlplane/handlers_providers.go [hunk 3]
```diff
@@ -108,7 +108,7 @@ func (s *Server) ListProviders(ctx context.Context, req *minderv1.ListProvidersR
 		return nil, err
 	}
 
-	zerolog.Ctx(ctx).Debug().Int("count", len(list)).Msg("providers")
+	zerolog.Ctx(ctx).Debug().Int("count", len(list)).Msg("ghProviders")
```
Comment (eleftherias) [inline internal/controlplane/handlers_providers.go:111]:
```suggestion
	zerolog.Ctx(ctx).Debug().Int("count", len(list)).Msg("providers")
```

File: internal/controlplane/handlers_providers.go [hunk 4]
```diff
@@ -151,7 +151,7 @@ func (s *Server) ListProviders(ctx context.Context, req *minderv1.ListProvidersR
 func (_ *Server) ListProviderClasses(
 	_ context.Context, _ *minderv1.ListProviderClassesRequest,
 ) (*minderv1.ListProviderClassesResponse, error) {
-	// Note: New provider classes should be added to the providers package.
+	// Note: New provider classes should be added to the ghProviders package.
```
Comment (eleftherias) [inline internal/controlplane/handlers_providers.go:154]: more rename in comments

File: internal/controlplane/handlers_repositories.go [hunk 5]
```diff
@@ -372,9 +372,9 @@ func (s *Server) ListRemoteRepositoriesFromProvider(
 		out.Results = append(out.Results, results...)
 	}
 
-	// If all providers failed, return an error
+	// If all ghProviders failed, return an error
```
Comment (eleftherias) [inline internal/controlplane/handlers_repositories.go:375]:
```suggestion
	// If all providers failed, return an error
```

File: internal/controlplane/handlers_repositories.go [hunk 6]
```diff
@@ -470,11 +470,11 @@ func (s *Server) inferProviderByOwner(ctx context.Context, owner string, project
 	}
 	opts, err := s.providerStore.GetByNameAndTrait(ctx, projectID, providerName, db.ProviderTypeGithub)
 	if err != nil {
-		return nil, fmt.Errorf("error getting providers: %v", err)
+		return nil, fmt.Errorf("error getting ghProviders: %v", err)
```
Comment (eleftherias) [inline internal/controlplane/handlers_repositories.go:473]:
```suggestion
		return nil, fmt.Errorf("error getting providers: %v", err)
```

**Review by eleftherias — CHANGES_REQUESTED** [review]
Makes sense. It seems like the IDE rename took it too far and started renaming providers -> ghProviders in comments and strings. I flagged a few but there are more.

File: internal/controlplane/handlers_repositories_test.go [hunk 7]
```diff
@@ -147,7 +147,7 @@ func TestServer_ListRemoteRepositoriesFromProvider(t *testing.T) {
 		{
 			Name:          "List remote repositories fails when all providers error",
 			GitHubSetup:   newGitHub(withFailedListAllRepositories(errDefault)),
-			ExpectedError: "cannot list repositories for providers: [github]",
+			ExpectedError: "cannot list repositories for ghProviders: [github]",
```
Comment (eleftherias) [inline internal/controlplane/handlers_repositories_test.go:150]: One more snuck in

File: internal/providers/github/service/service.go [hunk 8]
```diff
@@ -13,7 +13,8 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package providers
+// Package service contains the github
```
Comment (dmjb) [inline internal/providers/github/service/service.go:16]: Created a separate package to avoid an import cycle

**Review by eleftherias — APPROVED** [review]

**Pull request merged**